### PR TITLE
[#11003] Add GitHub action for welcoming new contributors

### LINKED ISSUE #11003: Add GitHub Action to welcome new users
It would be good to display an automated message especially aimed at new contributors (first time issue reporters, and especially first time PR submissions) who might feel a little bit like they're posting into a black hole, just to let them know that their message was received, and to give them instructions on how to escalate the issue if they feel it's being neglected.

I would also like that such a message @'s me personally, so that I can be made more easily aware of new contributors and can follow up with them, and also so that they have an initial point of contact (as it is I am having trouble keeping up with new issues but this would help).

I have discussed this idea before as a possible enhancement to astropy-bot, but it seems this is also doable with GitHub Actions, and could also be deployed to other projects that wish it (e.g. adding it to package-template).  I.e. in the same vein as #10974.  In fact I was reading the docs on Actions (which I have still not tried using on any projects yet) and it listed this explicitly as a use case for them, so it should require very little effort IIUC.

### PR DESCRIPTION
**Pull request by embray**
Fixes #11003.

@pllim Any ideas for how we can test this?  Or should we just assume it works and wait to see if a new user posts an issue (or maybe I can create a dummy account to test it).  Alternatively, I can merge it on my fork of astropy and see if it works on forks too...

Suggestions very welcome on how to improve the text of the messages.

### CODE REVIEW

**Comment by pllim** [comment]
Looks great at a glance but I'll do a detailed review later.

I can test this next week after the long weekend. I have a special fork for such things. 😸  Thanks for your patience!

**Comment by embray** [comment]
Should I submit an issue to your testing fork to test it? =D

**Comment by pllim** [comment]
I wouldn't say you "should" but you "can" but I haven't gotten to it yet, I am still stuck in the Mountain of Madness (email pile).

File: .github/workflows/greetings.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            Code](https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md#checklist-for-contributed-code)
+            and make sure you've addressed as many of the questions there as
+            possible.
+
+            If you feel that this pull request has not been responded to in a
+            timely manner, please leave a comment mentioning our software
+            support engineer @embray, or send a message directly to the
+            [development mailing
+            list](http://groups.google.com/group/astropy-dev).  If the issue is
+            urgent or sensitive in nature (e.g. a security vulnerability)
```
Comment (pllim) [inline .github/workflows/greetings.yml:38]:
```suggestion
            urgent or sensitive in nature (e.g., a security vulnerability)
```

**Review by pllim — COMMENTED** [review]
Okay, I deployed it on my fork (https://github.com/kali-maa/astropy/blob/300973f420f63dc1289774968560f2cd77a3bb8f/.github/workflows/open_actions.yml#L20).

Feel free to test it out there!

**Comment by embray** [comment]
Ah, I just saw your review.  I'll have a look now.

File: .github/workflows/greetings.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: "Greeting message to new contributors"
```
[the remark this answers is not in the document]
Comment (embray) [reply .github/workflows/greetings.yml:1]: Yes, I see you were doing some other related experiments in your fork.

File: .github/workflows/greetings.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+on: ["issues", "pull_request_target"]
```
[the remark this answers is not in the document]
Comment (embray) [reply .github/workflows/greetings.yml:2]: In the case of the first-interaction actions I was just going by what it says in the docs to do.  I was curious about this well and I saw that it has an explicit check that the type is "opened".  So there is also no reason for the action to run at all without type="opened" specified explicitly in "on".

**Comment by embray** [comment]
@pllim Addressed your comments and made some additional fixes (namely, so that the message doesn't all run together into a single paragraph)

**Review by pllim — APPROVED** [review]
LGTM now. Thanks!

Shall we merge? 😸

**Pull request merged**